### DISCUSSION TITLE
Can someone explain a little more about how the load/unload works?

### DISCUSSION THREAD

**Discussion by drf-jpg** [Opinionated-Blueprints/10-Books-Full-of-Rails#71] [Q&A] [answered]
I have a few different questions about the loading/unloading stations. Some of them are related to these particular blueprints while others are more generic. I've read the Wiki, but I wasn't able to get detailed enough explanations.

1. For the requestor stations, what exactly is the difference between the '2/wagon' and '3/wagon' blueprints?
2. For the requestor stations, what is the use case for the '3x parallel output addon' and '4x parallel output addon'? What do they accomplish?
3. For the provider stations, what's the high-level "algorithm" for assigning trains with mixed cargo?
4. For both the provider and requestor stations, how are the inserters paired with the belts? Where did that come from? 

For question 4 specifically, let me expand on what I'm trying to ask...
- Look at the blueprint for loading a train (Providers > 1 Car).
- I've pasted a screenshot of the current description below.

![image](https://github.com/user-attachments/assets/0f0b8258-12a5-4d60-bc5a-e001fbd49958)

- What were the calculations to determine the pairing of inserter-to-belt? Like, it looks like this blueprints works with blue inserters for both red and blue belts, but if you want to use green belts it requires green inserters. How was that computed? Is that optimal or something?
- What does the 'stack' icon mean next to an inserter? For example, when it says 'green belts x3 stacks' it requires the stack inserter instead. What does that stack icon represent here?

**Comment by Bocian-1** [comment, marked as answer]
1. This is how many fully saturated belts this setup can get from a wagon or side in case of the 1/side BPs

2. The main (1/side(L) and 1/side(R)) unloaders are compact enough that you can run belts parallel to the tracks for up to 4 of them, but routing them while simple, doing it with a blueprint is just faster. These unloaders also can't be flipped due to how inserters choose the side of the belt to place the item on. So, these parallel output addons are that parallel routing of belts you can place on top like so:
![image](https://github.com/user-attachments/assets/dc2c7aa8-d660-49fe-b73a-d33b5a2ad684)

3. You just load mixed cargo into the train preferably with a balancer to load wagons somewhat evenly. If a train has any cargo it selects one in some order (not random, but it doesn't really matter for us here) and goes to a requester for that item. Thanks to filtered inserters on requesters only the correct item will be unloaded and if the train still has some cargo it will just continue on its way to another requester requesting one of the items in the train's cargo.

4. There was no math. I just have a few setups like this and used increasingly faster inserters to check for the cheapest options that can keep up and get the belt moving at full capacity. I used a circuit-based belt throughput calculator to make sure they're running at full throughput.
![image](https://github.com/user-attachments/assets/42d2daab-b02d-4d72-9159-feaca0596d2a)
The stack icons refer to the belt capacity
![image](https://github.com/user-attachments/assets/8edc2958-3022-4986-8e67-ad3b3bde5e6a)
For loading, bulk inserters can sometimes manage as it's all about the hand size and swing speed but for unloading only stack inserters can place stacks of items.
